### ISSUE TITLE
make grobid skip pdf files

### ISSUE THREAD

**Issue by MariusKempf** [kermitt2/grobid#582]
Hi everyone,

I have got a huge bunch of pdf-files which I would like to process. Some of them are not working (I guess because of some weird pdf-format). Is there a way I can make grobid skip a file? For example if it takes to long?
Unfortunally there is no way to identify the problematic files in advance.

**Comment by lfoppiano** [comment]
Hi @MariusKempf, 
   AFAIK if one file is failing, grobid should tell you what went wrong and move one the next file. How are you processing them? 

If you need to process them in parallel, you can you spawn grobid as a server and use the [grobid-python-client](https://github.com/kermitt2/grobid-client-python) via the API, it should also automatically skip problematic files. 

Regarding long-processing files, Grobid already has a system to avoid processing too-big files (e.g there is a timeout on the conversion from pdf to XML and some limit on the maximum numbers of blocks and tokens a PDF can have).

**Comment by kermitt2** [comment]
As a complement to Luca answer, there are already 3 mechanisms to skip PDF, in the sense "stop working on this PDF when it turns bad": 

- a timeout to the call to pdfalto, pdfalto is doing the low level pdf parsing (the default value is 60s, we're generous, it can be changed in `grobid/grobid-home/config/grobid.properties`)

- a memory limit to the process running pdfalto, so we are sure to avoid having swap, JVM disaster, etc. (the default value is 6096MB, we're generous, it can be changed in `grobid/grobid-home/config/grobid.properties`)

- a maximum block number and a maximum token number (they can be changed in `grobid/grobid-home/config/grobid.properties`)

Nothing else because we really have to at least parse the PDF to see if it is weird. We can have gentle PDF of 200MB because they have big embedded images, and nasty well-looking lightweight PDF.

**Comment by MariusKempf** [comment]
Hi @kermitt2 and @lfoppiano 
Thank you guys a lot - this helped me to fix quite some things on how I integrate your tool into my project.
I still have problems parsing some PDF files, i.e.:
[problem_7b6fe606-5570-4024-aca6-7481fafa39b7.pdf](https://github.com/kermitt2/grobid/files/4661851/problem_7b6fe606-5570-4024-aca6-7481fafa39b7.pdf)
[problem_d052c049-bf4d-49b0-9d40-065208284090.pdf](https://github.com/kermitt2/grobid/files/4661853/problem_d052c049-bf4d-49b0-9d40-065208284090.pdf)
Those might some examples which can´t even be parsed to find out if they are 'weird' like you said.

Unfortunately, Grobid is not stopping or skipping them and my crawler can´t separate them during crawling.

**Comment by lfoppiano** [comment]
Hi @MariusKempf, 
mmmm .... that's weird because I've just tried and I've got immediately an answer. 

[Archive.zip](https://github.com/kermitt2/grobid/files/4662463/Archive.zip)

I wonder what's happening with your crawler? How did you implement it? Do you call grobid via API or in a different way? What's happening? it is just hanging?

**Comment by MariusKempf** [comment]
Hi @lfoppiano 
Thanks for your efforts! yes, it´s very weird.
I just cloned the grobid-service and the python-client again from your repository to make sure everything is up-to-date. Furthermore, I changed Java back to JDK8.
I am using the local server on default port etc. and the python client. All my other example files are working just those two keep grobid hanging/busy

The problem seems to be connected with Win(10) since on my Linux server it´s working without any problems and amazingly fast as well. On Win it always gets stuck or hanging when it gets to those two files.
